common/src/Model/Brush.cpp

```diff
@@ -947,8 +970,23 @@ namespace TrenchBroom {
             assert(remaining.vertexCount() + moving.vertexCount() == vertexCount());
             
             // Will vertices be removed?
-            if (!allowVertexRemoval && result.vertexCount() < m_geometry->vertexCount())
-                return false;
+            if (!allowVertexRemoval) {
+                if (result.vertexCount() < m_geometry->vertexCount())
```

Comment:
One example: you could remove the selected vert in this screenshot:
<img width="415" alt="screen shot 2016-12-27 at 2 33 08 am" src="https://cloud.githubusercontent.com/assets/239161/21496592/e334a2fa-cbdc-11e6-8b32-bcc663508add.png">
by moving the top face like this:
<img width="323" alt="screen shot 2016-12-27 at 2 35 58 am" src="https://cloud.githubusercontent.com/assets/239161/21496653/3ef97f52-cbdd-11e6-89a4-07577c6959c6.png">

This is the test case of `BrushTest.movePolygonRemainingPolyhedron` and the line `checkCanNotMoveTopFace(brush, Vec3(-32.0, -32.0, 0.0)); // Causes face merging and a vert to be deleted at z=-64` attempts that face move.

This is denied by the `(result.vertexCount() < m_geometry->vertexCount())` check, though.